File: src-tauri/src/commands/helm_repositories.rs
//! Repository configuration belongs to the local Helm installation, not a cluster.
use crate::{error::AppResult, k8s::helm_cli};

#[tauri::command]
pub async fn helm_repository_list() -> AppResult<Vec<helm_cli::HelmRepository>> {
    helm_cli::list_repositories().await
}

#[tauri::command]
pub async fn helm_repository_add(name: String, url: String) -> AppResult<()> {
    helm_cli::add_repository(&name, &url).await
}

#[tauri::command]
pub async fn helm_repository_remove(name: String) -> AppResult<()> {
    helm_cli::remove_repository(&name).await
}

#[tauri::command]
pub async fn helm_repository_update() -> AppResult<()> {
    helm_cli::update_repositories().await
}

File: src/lib/helmRepositories.ts
import { invoke } from "@tauri-apps/api/core";

export interface HelmRepository {
  name: string;
  url: string;
}

/** Uses the local Helm CLI configuration shared by all clusters on this machine. */
export const helmRepositories = {
  list: () => invoke<HelmRepository[]>("helm_repository_list"),
  add: (name: string, url: string) =>
    invoke<void>("helm_repository_add", { name, url }),
  remove: (name: string) => invoke<void>("helm_repository_remove", { name }),
  update: () => invoke<void>("helm_repository_update"),
};
